From 38831e4ea149a0b4731d123c63d8b493d30ad0be Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Sat, 26 May 2018 08:16:02 +0200
Subject: [PATCH] udisks2: Fix crashes caused by missing source tag

GAsyncReadyCallback is never called from g_drive_stop, because
source_tag is not set, but checked. This obviously causes issues
for client applications. Add missing source_tag.

Closes: https://gitlab.gnome.org/GNOME/gvfs/issues/1
---
 monitor/udisks2/gvfsudisks2drive.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/monitor/udisks2/gvfsudisks2drive.c b/monitor/udisks2/gvfsudisks2drive.c
index 52e9b75e..87656688 100644
--- a/monitor/udisks2/gvfsudisks2drive.c
+++ b/monitor/udisks2/gvfsudisks2drive.c
@@ -915,6 +915,7 @@ gvfs_udisks2_drive_stop (GDrive              *_drive,
   GTask *task;
 
   task = g_task_new (drive, cancellable, callback, user_data);
+  g_task_set_source_tag (task, gvfs_udisks2_drive_stop);
 
   /* This information is needed in GVfsDdisks2Volume when apps have
    * open files on the device ... we need to know if the button should
-- 
2.23.0

